src/data/polynomial/laurent.lean

```diff
@@ -294,6 +307,46 @@ left_inverse_trunc_to_laurent.injective
   f.to_laurent = g.to_laurent ↔ f = g :=
 ⟨λ h, polynomial.to_laurent_injective h, congr_arg _⟩
 
+lemma exists_T_pow (f : R[T;T⁻¹]) :
+  ∃ (n : ℕ) (f' : R[X]), f'.to_laurent = f * T n :=
+begin
+  apply f.induction_on' _ (λ n a, _); clear f,
+  { rintros f g ⟨m, fn, hf⟩ ⟨n, gn, hg⟩,
+    by_cases h : m ≤ n;
+    refine ⟨m + n, fn * X ^ n + gn * X ^ m, _⟩;
+    simp only [hf, hg, add_mul, add_comm (n : ℤ), map_add, map_mul, polynomial.to_laurent_X_pow,
+      mul_T_assoc, int.coe_nat_add] },
+  { cases n with n n,
+    { exact ⟨0, polynomial.C a * X ^ n, by simp⟩ },
+    { refine ⟨n + 1, polynomial.C a, _⟩,
+      simp only [int.neg_succ_of_nat_eq, polynomial.to_laurent_C, int.coe_nat_succ, mul_T_assoc,
+        add_left_neg, T_zero, mul_one] } }
+end
+
+/--  This version of `exists_T_pow` can be called as `rcases f.exists_T_pow' with ⟨n, f', rfl⟩`. -/
+lemma exists_T_pow' (f : R[T;T⁻¹]) :
```

Comment:
This might be nice to phrase as a recursor like the one below; that way it works with the `induction` tactic

Reply:
That is:
```suggestion
lemma induction_on_mul_T (f : R[T;T⁻¹]) {Q : R[T;T⁻¹] → Prop}
  (Qf : ∀ (f : R[X]) (n), Q (f.to_laurent * T n)) :
  Q f :=
```

Reply:
I used your version, but I never know how to use the `induction` tactic.  For instance, is there a better way to use this lemma in the proof of `reduce_to_polynomial_of_mul_T`?

I used `refine f.induction_on_mul_T (λ f n, _),` avoiding `induction`.

Reply:
You should be able to use
```
induction f using laurent_polynomial.induction_on_mul_T with f n
```

Reply:
Thanks: it works!

I added `elab_as_eliminator` (I did not try without) and had to switch the order of assumptions in `induction_on_mul_T` to get it to work.